modules/imgcodecs/perf/perf_decode_encode.cpp

```diff
@@ -11,6 +11,80 @@ namespace opencv_test
 
 using namespace perf;
 
+static Animation makeCirclesAnimation(Size size = Size(320, 240), int type = CV_8UC4, int nbits = 8, int frameCount = 40)
+{
+    struct AnimatedCircle {
+        cv::Point2f pos;
+        cv::Point2f velocity;
+        float radius;
+        float radius_speed;
+        cv::Scalar color;
+        cv::Scalar border_color;
+    };
+
+    const int numCircles = 80;
+    const int maxval = (1 << nbits) - 1;
+
+    cv::RNG rng = theRNG();
+    std::vector<AnimatedCircle> circles;
+    Animation animation;
+
+    // Initialize animated circles
+    for (int i = 0; i < numCircles; ++i) {
+        AnimatedCircle c;
+        c.pos = cv::Point2f(rng.uniform(0.f, (float)size.width),
+            rng.uniform(0.f, (float)size.height));
+        c.velocity = cv::Point2f(rng.uniform(-2.f, 2.f),
+            rng.uniform(-2.f, 2.f));
+        c.radius = rng.uniform(10.f, 40.f);
+        c.radius_speed = rng.uniform(-0.5f, 0.5f);
+        c.color = cv::Scalar(rng.uniform(0, maxval),
+            rng.uniform(0, maxval),
+            rng.uniform(0, maxval),
+            rng.uniform(230, maxval));
+        c.border_color = c.color;
+        circles.push_back(c);
+    }
+
+    // Generate frames
+    for (int frame = 0; frame < frameCount; ++frame) {
+        cv::Mat img(size, type, cv::Scalar(20, 0, 10, 128));
+
+        for (size_t i = 0; i < circles.size(); ++i) {
+            AnimatedCircle& c = circles[i];
+
+            // Update position
+            c.pos += c.velocity;
+
+            // Bounce on edges
+            if (c.pos.x < 0 || c.pos.x > size.width) c.velocity.x *= -1;
+            if (c.pos.y < 0 || c.pos.y > size.height) c.velocity.y *= -1;
+
+            // Update radius
+            c.radius += c.radius_speed;
+            if (c.radius < 10.f || c.radius > 80.f) {
+                c.radius_speed *= -1;
+                c.radius = std::max(10.f, std::min(c.radius, 80.f));
+            }
+
+            c.color = c.color - Scalar(c.velocity.x, 0, c.velocity.y, rng.uniform(1, 4));
+
+            // Draw
+            cv::circle(img, c.pos, (int)c.radius, c.color, cv::FILLED, cv::LINE_AA);
+            cv::circle(img, c.pos, (int)c.radius, c.border_color, 1, cv::LINE_AA);
+        }
+
+        animation.frames.push_back(img.clone());
```

Comment:
Img is local variable inside for loop body and re-created on each loop iteration. No need to call clone for it.